### PR TITLE
Fix up and modify some bits of the documentation

### PR DESCRIPTION
**Pull request by anoadragon453**
Some documentation fixes justified in github comments. Overall this documentation is excellent. I haven't tried converting/writing a test yet, but the information already available is very helpful.

### CODE REVIEW

File: ONBOARDING.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ If you've not run Complement tests yet, please do so. This document will outline
...
+Complement runs Docker containers every time you call `deployment := must.Deploy(...)`, which gets killed on `deployment.Destroy(...)`. These containers are snapshots of the target homeserver at a particular state. The state is determined by the `Blueprint`, which is a human-readable outline for what should be done prior to the test. Coming from Sytest, a `Blueprint` is similar to a `fixture`. A `deployment` has functions on it for creating deployment-scoped structs such as CS-API clients for interacting with specific Homeservers in the deployment. Assertions are done via the `must` and `match` packages. For testing outbound federation, Complement implements a bare-bones Federation server for homeservers to talk to. Unlike Sytest, you have to explicitly opt-in to attaching core functionality to the server so the reader can clearly see what is and is not being handled automatically. This is done using [functional options](https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis) and looks something like:
```
Comment (anoadragon453) [inline ONBOARDING.md:7]: Perhaps we should enforce a documentation line length :) FYI this was changing:

* Homeserver -> homeserver
* CSAPI -> CS-API

File: ONBOARDING.md [hunk 2]
```diff
@@ -16,18 +16,20 @@ srv := federation.NewServer(t, deployment,
...
+|      | :12345 <------ host.docker.internal:12345 ---- |           |                                                       |           |
 | Host |                                                | Container | ------- SS API https://hs2/_matrix/federation/... --> | Container |
-|      | -------CS API http://localhost:38295 --------> |   (hs1)   |                                                       |   (hs2)   |
-|      | ------SS API https://localhost:37562 --------> |           |                                                       |           |
+|      | -------CS API http://localhost:54321 --------> |   (hs1)   |                                                       |   (hs2)   |
+|      | ------SS API https://localhost:55555 --------> |           |                                                       |           |
 +------+                                                +-----------+                                                       +-----------+
-                                             docker -p 38295:8008 -p 37562:8448
+                                             docker -p 54321:8008 -p 55555:8448
```
Comment (anoadragon453) [inline ONBOARDING.md:32]: I found these simple numbers easier to match up with each other than the random ones.

File: README.md [hunk 3]
```diff
@@ -15,9 +15,9 @@ You can either use your own image, or one of the ones supplied in the [dockerfil
 For instance, for Dendrite:
 ```
 # build a docker image for Dendrite...
-(cd dockerfiles && docker build -t complement-dendrite -f Dendrite.Dockerfile .)
+$ cd dockerfiles && docker build -t complement-dendrite -f Dendrite.Dockerfile .
```
Comment (anoadragon453) [inline README.md:18]: afaik this didn't need `()`'s

File: ONBOARDING.md [hunk 4]
```diff
@@ -79,7 +81,7 @@ func TestFoo(t *testing.T) {
 }
 ```
 
-This slightly awkward structure means `sytest_coverage.go` will know the test is converted and automatically update the list when run! If you are struggling to get the script to pick up the file, just do `go run sytest_coverage.go -v` to see the exact filename and expected string.
+This slightly awkward structure means `sytest_coverage.go` will know the test is converted and automatically update the list in the README when run! If you are struggling to get the script to pick up the file, just do `go run sytest_coverage.go -v` to see the exact filename and expected string.
```
Comment (Kegsay) [inline ONBOARDING.md:84]: It doesn't automatically update the README, I have to copypasta!

Comment (anoadragon453) [reply ONBOARDING.md:84]: Oh, unfortunate :)

Comment (Kegsay) [inline ONBOARDING.md:84]:
```suggestion
This slightly awkward structure means `sytest_coverage.go` will know the test is converted and automatically update the list when run! If you are struggling to get the script to pick up the file, just do `go run sytest_coverage.go -v` to see the exact filename and expected string.
```

File: README.md (hunk 3 above)
Comment (Kegsay) [reply README.md:18]: It doesn't but the subsequent command has to be run from the parent directory. Removing the parentheses breaks the next command.

Comment (Kegsay) [inline README.md:18]:
```suggestion
$ (cd dockerfiles && docker build -t complement-dendrite -f Dendrite.Dockerfile .)
```

Comment (anoadragon453) [reply README.md:18]: Oh, nice trick!

**Comment by anoadragon453** [comment]
@Kegsay Updated! I'm not able to merge myself though I'm afraid :)

**Pull request merged**